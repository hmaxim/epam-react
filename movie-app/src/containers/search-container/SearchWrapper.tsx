import styled from "styled-components";

const searchWrapper = styled.div`
  height: 25%;
  width: 100%;

  .search-container {
    padding: 0 9.5%;

    .search-title {
      color: #ffffff;
      letter-spacing: 1.5px;
      margin-bottom: 10px;
      vertical-align: top;
    }

    button {
      width: 150px;
      height: 39.5px;
      font-size: 20px;
      vertical-align: top;
    }

    .search-buttons-container {
      display: flex;
      align-items: center;
      color: white;
      padding: 20px 0;

      span {
        letter-spacing: 1px;
        margin-right: 30px;
        font-size: 20px;
      }

      button {
        width: auto;
        border: none;
        cursor: pointer;
      }
    }
  }
`;

export default searchWrapper;
