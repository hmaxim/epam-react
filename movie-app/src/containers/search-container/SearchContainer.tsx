import React, { useState } from "react";
import SearchWrapper from "./SearchWrapper";
import Input from "../../shared-components/input/InputWrapper";
import ButtonWrapper from "../../shared-components/button/ButtonWrapper";
import SwitchersButtons from "../../shared-components/switchers-buttons/SwitchersButtons";

const searchButtonsParams = [
  { label: "Title", active: true, buttonValue: "title" },
  { label: "Genre", active: false, buttonValue: "genre" }
];

const SearchContainer = () => {
  return (
    <SearchWrapper>
      <div className="search-container">
        <h1 className="search-title">FIND YOUR MOVIE</h1>
        <div>
          <Input></Input>
          <ButtonWrapper>SEARCH</ButtonWrapper>
          <SwitchersButtons switchersTitle={'SEARCH BY'}
          searchButtonsParams={searchButtonsParams}></SwitchersButtons>
        </div>
      </div>
    </SearchWrapper>
  );
};

export default SearchContainer;
