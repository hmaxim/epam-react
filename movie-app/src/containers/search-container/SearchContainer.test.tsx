import 'jsdom-global/register';
import React, { Component } from 'react';
import { shallow, mount, render } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import SearchContainer from './SearchContainer';
import SearchWrapper from './SearchWrapper';

describe('SearchContainer', () => {
  let component: any;

  beforeEach(() => {
    component = render(<SearchContainer />);
  });

  test('to be render', () => {
      expect(component.find(SearchWrapper)).toMatchSnapshot()
  });
});
