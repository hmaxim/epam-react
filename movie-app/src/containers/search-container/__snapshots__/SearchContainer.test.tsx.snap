// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer to be render 1`] = `null`;
